api/v1beta1/disruption_webhook_test.go

```diff
@@ -252,6 +252,74 @@ var _ = Describe("Disruption", func() {
 			})
 		})
 
+		Describe("spec.trigger possible errors", func() {
+			BeforeEach(func() {
+				k8sClient = makek8sClientWithDisruptionPod()
+				deleteOnly = false
+			})
+
+			JustBeforeEach(func() {
+				newDisruption = makeValidNetworkDisruption()
+				controllerutil.AddFinalizer(newDisruption, chaostypes.DisruptionFinalizer)
+			})
+
+			AfterEach(func() {
+				k8sClient = nil
+				newDisruption = nil
+			})
+
+			When("notBefore is set past duration", func() {
+				It("should return an error", func() {
+					newDisruption.Spec.Duration = "10m"
+					newDisruption.Spec.Trigger = &DisruptionTriggers{
+						Inject: &DisruptionTrigger{
+							NotBefore: metav1.NewTime(time.Now().Add(time.Minute * 15)),
+						},
+					}
+
+					err := newDisruption.ValidateCreate()
+
+					Expect(err).Should(HaveOccurred())
```

Comment:
Prefer:
```suggestion
					Expect(newDisruption.ValidateCreate()).Should(MatchError("you should not set spec.trigger.*.notBefore to a time farther in the future than the disruption duration"))
```

It guarantees error has occurred AND contains expected error text.

Reply:
You might want to replace all your tests by [Ginkgo table testing](https://onsi.github.io/ginkgo/#table-specs) (you can still use BeforeEach with that):

```suggestion
DescribeTable("spec.trigger possible errors", func(duration string, pods, inject *DisruptionTrigger, wantError string){
  newDisruption.Spec.Duration = duration
					newDisruption.Spec.Trigger = &DisruptionTriggers{
						Pods: pods,
						Inject: inject,
					}

  Expect(newDisruption.ValidateCreate()Should(MatchError(wantError))
},
Entry("notBefore is set past duration", "10m", nil, &DisruptionTrigger{
							NotBefore: metav1.NewTime(time.Now().Add(time.Minute * 15)),
						}, "you should not set spec.trigger.*.notBefore to a time farther in the future than the disruption duration"),
Entry("offset is set larger than duration", "10m", &DisruptionTrigger{
							NotBefore: metav1.NewTime(time.Now().Add(time.Minute * 15)),
						}, nil, "you should not set spec.trigger.*.offset higher than the disruption duration"))
```